### PR TITLE
[feature] Pass current user and current visit to api data pipeline

### PR DESCRIPTION
**Pull request by donrestarone**
## ability to reference `current_user` and `current_visit` in API Actions
Addresses: https://github.com/restarone/violet_rails/issues/457
Screenshot from 2022-07-11 17-57-56
## example usage
Screenshot from 2022-07-11 17-58-10

### CODE REVIEW

File: app/models/api_action.rb [hunk 1]
```diff
@@ -73,7 +74,7 @@ def redirect;end
   def serve_file;end
 
   def request_headers
-    headers = custom_headers.to_json.gsub('SECRET_BEARER_TOKEN', bearer_token)
+    headers = custom_headers_evaluated.gsub('SECRET_BEARER_TOKEN', bearer_token)
```
Comment (donrestarone) [inline app/models/api_action.rb:77]: @alis-khadka does this allow dynamic headers here? (see below) -- if so can you add in-line documentation
Screenshot from 2022-07-09 16-21-59

**Comment by donrestarone** [comment]
@alis-khadka can you attach some screenshots and a demo video? I believe in-line documentation for the variables in scope would be nice too!

File: app/models/api_action.rb [hunk 2]
```diff
@@ -26,6 +26,7 @@ class ApiAction < ApplicationRecord
   validates :http_method, inclusion: { in: ApiAction::HTTP_METHODS}, allow_blank: true
   
   validates :payload_mapping, safe_executable: true
+  validates :custom_headers, safe_executable: true
```
Comment (Pralish) [inline app/models/api_action.rb:29]: @alis-khadka , all dynamic attributes are safe by default. You don't need to validate it here. The validation for payload_mapping can also be removed.

**Comment by donrestarone** [comment]
## :spiral_notepad: UAT on Heroku underway

**Pull request merged**